### PR TITLE
Tinkering with BSM2 Optimization

### PR DESCRIPTION
**Pull request by adam-a-a**
## Summary/Motivation:
Started as a draft PR for the team to look into optimization of the BSM2 flowsheet. Problems could stem from poor scaling, poor initialization, problematic cost relationships (ran into ampl errors a couple times already for aeration basin capital cost constrint), or all of the above (or some other unknown reason).  

## Changes proposed in this PR:
- add setup_optimization function
- revise costing
- refactor cstr_injection and create AerationTank unit model class which inherits from cstr_injection 
- alter the cost method for cstr (without injection) to one that appears more practical

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py]: did this to avoid ampl error, where base of exponential term was going slightly negative.

File: watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py]: I also barely tried playing with effluent limits for TSS, COD, and total N.

**Comment by adam-a-a** [comment]
Note to self: clean this up and get it merged!

File: watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py
Comment (MarcusHolly) [inline watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py:383]: Does this need to be done here and in the initialization?

File: watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/test/test_full_WRRF_with_ASM1_ADM1.py
Comment (MarcusHolly) [inline watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/test/test_full_WRRF_with_ASM1_ADM1.py]: Delete?

**Review by MarcusHolly — APPROVED** [review]
LGTM - just a few minor questions. For some reason, that Linux 3.8 notebook test has been running into that same error for me today as well. Re-running it once or twice has resolved it

**Review by agarciadiego — APPROVED** [review]
LGTM - maybe add an issue after merge to try to get through the large amount of TODOs

**Review by luohezhiming — COMMENTED** [review]
Looks good to me, I think it is ready for merging, the Mac setup failed, I think it can be resolved by adding `@pytest.mark.requires_idaes_solver`

**Review by luohezhiming — APPROVED** [review]

File: watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py
Comment (adam-a-a) [reply watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py:383]: @MarcusHolly Yes, because the initializer for the mixer apparently reactivates the previously deactivated pressure equality constraints. So right now, we deactivate during the build/setting specifications and also after initializing.

**Pull request merged**
